% \documentclass[english, draft]{article}
\documentclass[english]{article}

\usepackage{geometry}
\usepackage{float}
\usepackage[utf8]{inputenc}
\usepackage[backend=biber,style= authoryear]{biblatex}
\usepackage[english]{babel}
\usepackage{csquotes}
\usepackage{graphicx}
\usepackage{subcaption}
\usepackage{booktabs}
\usepackage{xargs}
\usepackage[pdftex,dvipsnames,table]{xcolor}
\usepackage{bbold}
\usepackage{diagbox}
\usepackage{soul}

\graphicspath{{./resources/images}}
\addbibresource{../articles.bib}
\addbibresource{../manual.bib}

\geometry{a4paper, total={165mm,257mm}, left=15mm, top=20mm,}

\begin{document}
\pagenumbering{arabic}

\begin{abstract}

\end{abstract}

\section{Background}
European grapevine (Vitis vinifera L.) is one of the most economically and widespread worldwide crop. Around 33 of the  10.000 varieties,  cover 50\% of the world's vineyard surface (OIV, 2017).
Grapevine faces numerous pathogens and one of the most damaging diseases is downy mildew, caused by the oomycete Plasmopara viticola. This pathogen is endemic to the North America and was introduced into Europe around the 1870s  (Gessler et al., 2011), \parencite{fontaineEuropeBridgeheadWorldwide2021}). All the Vitis vinifiera varieties are susceptible to the pathogen and the current strategy to control the disease  relies on the massive applications of fungicides and which are potentially harmful for the humans and the environment. Long -lasting breeding programs have been conducted to introgress resistance from wild \textit{Vitis} species into elite resistant varieties~\parencite{merdinogluBreedingDurableResistance2018}, \parencite{topferCoolClimatePerspective2022}. More than 30 resistance genetic loci or \textit{Rpv} (Resistance to \textit{Plasmopara viticola}), have been already identified in grapevine (Maul,2021).  Most of them confer a partial resistance defined by various levels of sporulation and typology of necrosis or cell death. All of the \textit{Rpv} have been detected by analysis of macroscopic phenotypes of the disease and have been implemented by using several variables such as severity, incidence, spores count. Therefore, the use of the semi-quantitative OIV452~\parencite{OIVDescriptorList2009} scale was frequently used for assessing both sporulation and necrosis~\parencite{possamaiPhenotypingQTLIdentification2022},  Some \textit{Rpv} encode  intracellular nucleotide-binding leucine rich repeat receptors (NLRs) to detect pathogen effectors,  and activate effector triggered immunity (ETI) ETI is a robust immune response usually
accompanied by a form of programmed cell death known as the hypersensitive response or HR (Ngou et al., 2022).  One role of the HR is to limit the pathogen progression to the tissue region which is directly in contact with the pathogen (Balin-Kurti, 2019; Noman et al., 2020).  



\section{Material and Method}

The proposed plant method includes four main items: (i) The imaging acquisition system developed to acquire images of \textit{Vitis} leaf discs infected with \textit{Plasmopara viticola} used to create (ii) the dataset, which needs to benefit from (iii) pre-processing before investigating (iv) various approaches for the quantification of the interaction between the plant and the pathogen.

\subsection{Imaging system}

\begin{figure}[H]
    \begin{center}
        \includegraphics[width=0.5\linewidth]{2023_a_oiv_imaging_system.jpg}
        \caption{VEGOIA platform's manual imaging system.}\label{fig:vegoia}
    \end{center}
\end{figure}
The image system, as seen if Fig.~\ref{fig:vegoia}, consists of a copy stand and an RGB camera. The copy holds the camera at a specified height and has a Plexiglas with a hole on its base to allow constant framing, the lightning sources are used to remove any unwanted shadows from the images. Over the course of the years various cameras and lenses have bee used this causes the framing and exposure of the images to be inconsistent.

\subsection{Dataset}
Experiments to evaluate the resistance of various \textit{Vitis} species and genotypes to various strains of \textit{P. viticola} have been performed in the VEGOIA platform at INRAE center Grand Est Colmar since 2012. On these experiments, leaf discs extracted from various genotypes were infected with \textit{P. viticola}. Each leaf disc was scored with OIV 452-1 at 0 and 6 days post infection (dpi), some times also at 3, 4 and 5 dpi. Images of the plates containing the leaf discs were taken each time they were scored. Data regarding score, plate, position and image are kept in Excel files.
All data for training was extracted from experiments having similar acquisition settings that happened between 2020 and 2022.

\subsubsection{Leaf disc detection}
The pre-processing dataset consisted of 1932 plate images containing 22122 leaf discs sampled from the selected data. Ground truth was obtained with three steps: (i) 187 images were annotated with segmentation masks using the GiMP~\parencite{gimp}, these images were then used to train a segmentation model with ~\parencite{bergIlastikInteractiveMachine2019} used to create (iii) an object detection dataset.

\subsubsection{Annotated leaf discs}
\paragraph{}
\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{oiv1.png}
        \caption{OIV 1}\label{fig:oiv1}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{oiv3.png}
        \caption{OIV 3}\label{fig:oiv3}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{oiv5.png}
        \caption{OIV 5}\label{fig:oiv5}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{oiv7.png}
        \caption{OIV 7}\label{fig:oiv7}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{oiv9.png}
        \caption{OIV 9}\label{fig:oiv9}
    \end{subfigure}
    \caption{Examples of \textit{Vitis} leaf discs infected with \textit{Plasmopara viticola} annotated with OIV 452-1 values. Levels increase with resistance to pathogen. Upper row, susceptible leaf discs with levels 1 to 5. Lower row, resistant leaf disc~\ref{fig:oiv7} and fully resistant leaf disc~\ref{fig:oiv9}}\label{fig:oivlevels}. 
\end{figure}

\begin{table}[H]
    \centering
    \caption{Distribution of image quality and OIV values.}
    \label{tab:dataoivcount}
    \begin{tabular}{c|rrrrrrr}
        \toprule
        oiv & blur & color & crop & dark & good & water & total \\
        \midrule
        1   & 0    & 0     & 3    & 10   & 626  & 116   & 755   \\
        3   & 2    & 2     & 0    & 7    & 479  & 144   & 634   \\
        5   & 3    & 1     & 0    & 5    & 412  & 146   & 567   \\
        7   & 1    & 0     & 3    & 6    & 549  & 141   & 700   \\
        9   & 0    & 0     & 4    & 12   & 513  & 331   & 860   \\
        \bottomrule
    \end{tabular}
\end{table}

\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp21DM02_inoc2_T6_P35_a_4.png}
        \caption{Good image.}\label{fig:issuegood}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp20DM01_inoc2_T6_P01_c_3.png}
        \caption{Blurry image.}\label{fig:issueblur}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp21DM03_inoc3_T4_P32_c_1.png}
        \caption{Leaf with color issue.}\label{fig:issuecolor}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp21DM13_inoc2_T0_P34_b_1.png}
        \caption{Cropped image.}\label{fig:issuecrop}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp21DM04_inoc2_T6_P02_c_1.png}
        \caption{Dark image.}\label{fig:issuedark}
    \end{subfigure}
    \begin{subfigure}[b]{0.3\linewidth}
        \includegraphics[width=\linewidth]{Exp22DM01_inoc1_T5_P03_a_1.png}
        \caption{Image with water droplets.}\label{fig:issuewater}
    \end{subfigure}
    \caption{Examples of different issues present in the source images~\ref{fig:issuegood} represents a base good image. Some issues like darker images~\ref{fig:issuedark} can be fixed with pre-processing, others like color~\ref{fig:issuecolor}, crop~\ref{fig:issuecrop} and blur~\ref{fig:issueblur} are rare and can be discarded from the training . Finally images with water droplets~\ref{fig:issuewater} are very common enough that the model should be able to handle them.}\label{fig:imagequality}
\end{figure}

3516 leaf discs were sampled from the available images, for more information on the sampling method please see supplementary material, the dataset contains 531 different genotypes infected with eight different strains. All leaf discs were scored with an OIV 452-1 value, an example for each OIV 452-1 value can be seen in Fig.~\ref{fig:oivlevels}, as well as an image quality value, examples of image quality values can be seen in Fig.~\ref{fig:imagequality}, an overview of the annotation results can be seen in table~\ref{tab:dataoivcount}.

\paragraph{}
\begin{figure}[H]
    \centering
    \begin{subfigure}[b]{0.2\linewidth}
        \includegraphics[width=\linewidth]{p_viticola/resources/images/2023_a_oiv_sporulation.png}
        \caption{Sporulation}\label{fig:sporulation}
    \end{subfigure}
    \begin{subfigure}[b]{0.2\linewidth}
        \includegraphics[width=\linewidth]{p_viticola/resources/images/2023_a_oiv_stains.png}
        \caption{Flecks}\label{fig:flecks}
    \end{subfigure}
    \begin{subfigure}[b]{0.2\linewidth}
        \includegraphics[width=\linewidth]{p_viticola/resources/images/2023_oiv_a_dots.png}
        \caption{Dots}\label{fig:dots}
    \end{subfigure}
    \begin{subfigure}[b]{0.2\linewidth}
        \includegraphics[width=\linewidth]{p_viticola/resources/images/2023_a_oiv_senescence.png}
        \caption{Senscence}\label{fig:senescence}
    \end{subfigure}
    \caption{Examples of \textit{Vitis} leaf discs infected with \textit{Plasmopara viticola} showing the four different phenotypes selected.}
\end{figure}
\st{A subset of 1881 images were scored with the presence or absence of four phenotypes which are sporulation Fig.~\ref{fig:sporulation}, flecks Fig.~\ref{fig:flecks}, dots Fig.~\ref{fig:dots} and senescence Fig.~\ref{fig:senescence}. We believe that these four characteristics are enough to describe most of the observed variability in the images.}
\st{Due to the low frequency observed for some of the phenotypes we created the dataset by sampling the available leaf discs in ways that would yield the most images containing each phenotype. The detail of the sampling method can be seen in table the supplementary material. As an example we sampled 210 images with low hue values in the HSV color space to get images with potential senescence. Eighty five images were discarded due to poor quality.}
\begin{table}[H]
    \centering
    \caption{Distribution of annotations for binary phenotype detection.}
    \label{tab:databincount}
    \begin{tabular}{rrrr|r}
        \toprule
        sporulation &  necrosis\_dots &  necrosis\_stains &  necrosis\_senescence  & total\\
        \midrule
        1119        &             712 &               289 &                   444  & 1881\\
        \bottomrule
    \end{tabular}
\end{table}
Human annotation was performed by two experts. The results of the annotation can be seen in table~\ref{tab:databincount}.

\subsubsection{FUNDUR dataset}
A dataset of leaf discs scored with OIV 452-1 values using binocular magnifying glass from a subset of experiments of the FUNDUR project were added to validate the model on never seen images. These experiments aim to describe the interaction between 14 genotypes of \textit{Vitis} containing containing or not resistance factors rpv1, rpv3.1 or rpv10. The dataset contains 153 leaf discs scored with OIV 4552-1 values at DPIs 3,4,5 and 6.

\subsection{Pre-processing}

\begin{figure}[H]
    \begin{center}
        \includegraphics[width=0.9\linewidth]{p_viticola/resources/images/leaf_disc_extraction_2.png}
        \caption{Leaf disc extraction workflow.}\label{fig:preprocessing}
    \end{center}
\end{figure}

Since predictions are to be made on individual leaf discs they need to be extracted from the plates before being applied to the neural networks. Thus, the first step of pre-processing was to detect and index leaf discs. Indexation is needed to link the leaf discs with their genotypes and the strains used for infection.
Object detection will be handled by a pre-trained Fast R-CNN~\parencite{girshickFastRCNN2015} model. Leaf indexation will be performed with an image processing pipeline. Finally to prevent the model from taking the background into account when making predictions we then extract the largest square inscribe in the leaf disc which represents around 78\% of the full leaf disc. Figure~\ref{fig:preprocessing} shows a workflow of the leaf disc patch extraction and indexation from plates, which includes the steps described here below

\subsubsection{Leaf disc detection}
We selected a Fast R-CNN model, from the PyTorch library~\parencite{falcon2019pytorch}, pre-trained on COCO V1~\parencite{linMicrosoftCOCOCommon2015}. We selected a batch size of 20, the minimization was carried out using the ADAM optimizer~\parencite{kingmaAdamMethodStochastic2017} with a learning rate of $7\times 10^{-5}$ with decay using StepLR with step size of 10 and a gamma of 0.8. Data augmentation was performed with the albumentations library~\parencite{buslaevAlbumentationsFastFlexible2020} to increase data variability by changing exposure and using a horizontal flip.

\subsubsection{Leaf disc indexation}
Bounding boxes predicted by the object detector model will be used to index the leaf discs. Variations in framing and quantity of leaf discs in the plates make it so a trivial approach will not work.
In a first step leaf disc's row and column will be predicting with the centroid data from the bounding boxes applied to a mean shift~\parencite{comaniciuMeanShiftRobust2002a} for the columns and one for the rows. The varied positions of the plate combined to the possible absence of columns of leaf discs will cause the prediction the fail sometimes, this problem will only arise when less than ten leaf discs are present in a plate and the mean shift detects less than four columns. To address the first problem we used the Hough line detection~\parencite{dudaUseHoughTransformation1972} to detect the left border of the plate and crop the image to prevent the system for mistaken a plate aligned to the left with a missing first row. The second problem arises from the possibility of missing rows of leaf discs, to prevent this issue, when there are less than ten leaf discs detected, whenever there is enough space between two columns to fit 1.1 leaf discs, the next columns indexes are increased by one.

\subsubsection{Patch extraction}
In~\parencite{leeNewPerspectivesPlant2020} the authors show that deep learning models may take into account the image background when making predictions. To prevent this phenomena we decided to instead of the full leaf disc, just use the biggest square inscribed in the leaf disc which already represents around 78\% percent of the full surface.

\subsection{Deep learning methods}
The OIV 452-1 scale is a discrete scale with  an ordinality property. We tested different supervised strategies that benefit from this prior knowledge against the manually established ground truth.

\subsubsection{Swin transformer with cross-entropy loss}
As a naive baseline approach we used a Swin transformer~\parencite{liuSwinTransformerHierarchical2021} backbone, precisely we used Swin-T pretrained on ImageNet-1K, no ordinal prior was used with this model. Given a trainning set including K pairs of images x\textsubscript{i} and labels $\hat{y}\textsubscript{i}$, we trained the parameters $\theta$ of the network $f$ using stochastic gradient descent to minimize empirical risk
\begin{equation}
    \theta^*=arg~\min_{\theta} \sum_{i=1}^K\mathcal{L}(\hat{y_i}, f(x_{i}\theta))\label{fml:crossentropy}
\end{equation}
where $\mathcal{L}$ denotes the loss function, which was chosen as cross-entropy in our case. The minimisation was carried out using the ADAM optimizer with a learning rate of $1,6\times 10^{-4}$.

\subsubsection{Swin transformer with CORN loss}
The cross-entropy loss model is naive because it does not incorporate any prior knowledge of the ordinality of the OIV 452-1 scale. To improve the previous naive model, we substitute the cross-entropy loss with a rank-consistent ordinal regression framework (CORN)~\parencite{shiDeepNeuralNetworks2023}


% Annotation a décrire dans le texte et parler des divergences et accords garder trace des désaccords et en parler lors de la discussion -> ML approaches can stabilise?
% Enlever tous les désaccords avant le modèle? : Combien de données restent peux-t'on entrainer uniquement avec ça
% Enlever seulement une partie des données fausse le but de l'expérience
% Quelle question on pause quel dataset pour y répondre
% Est-ce que les erreurs sont dans les images difficiles ?
% Matrice de confusion avec ou sans désacord
% Critique de l'OIV ? 7 & 9 neccessaires


% Tables & figures
% Légendes
% Fontes
% Nom de colonnes


\begin{figure}[H]
    \centering
    \includegraphics[width=0.9\linewidth]{p_viticola/resources/images/2023_a_oiv_bin_corn.png}
    \caption{Rank-consistent ordinal regression (CORN) architecture with output example predicting OIV 7.}
    \label{fig:corn}
\end{figure}

\begin{equation}
    f(x^{[i]}) = \hat{P}(y^{[i]} > r_{k}|y^{[i]} > r_{k-1})\label{fml:binclass}
\end{equation}

\begin{equation}
    \hat{P}(y^{[i]} > r_{k}) = \prod_{j=1}^{k}f_{j}(x^{[i]})\label{fml:unconditionalprob}
\end{equation}

\begin{equation}
    OIV = (\sum_{j=1}^{K-1}\mathbb{1}(\hat{P}(y^{[i]} > r_{j}) > 0.5))*2 + 1\label{fml:rankprob}
\end{equation}

\section{Results and Discussion}

\subsection{Leaf Disc Detection}
\begin{table}[H]
    \centering
    \caption{Leaf disc detection models evaluation}
    \label{tab:leafdiscdetectionresult}
    \begin{tabular}{ll}
        \toprule
        scheduler & validation loss \\
        \midrule
        None &  0.017±0.002 \\
        steplr &  0.015±0.002 \\
        \bottomrule
    \end{tabular}
\end{table}

\subsection{Binary Predictions}

Phenotypes legend:
\begin{itemize}
    \item sp = sporulation
    \item nd = necrosis dots
    \item nf = necrosis flecks
    \item ns = necrosis stains
\end{itemize}

\begin{table}[H]
    \centering
    \caption{Results for binary pheotype detection.}
    \label{tab:binres}
    \begin{tabular}{lrl}
        \toprule
        phenotypes           & max training epochs & F1 weighted average \\
        \midrule
        sp, nd, nf, ns       & 5                   & 0.82\textpm0.031    \\
        sp, nd, nf, ns       & 10                  & 0.86\textpm0.007    \\
        sp, nd, nf, ns       & 20                  & 0.87\textpm0.010    \\
        sp, nd, nf, ns       & 200                 & 0.88\textpm0.008    \\
        \bottomrule
    \end{tabular}
\end{table}

\subsection{OIV prediction}

\begin{table}[H]
    \centering
    \caption{Results for OIV 452-1 ordinal regression.}
    \label{tab:oivres}
    \begin{tabular}{rlll}
        \toprule
        backbone max training epochs & backbone phenotypes  & mae              & mse              \\
        \midrule
        5                            & sp, nd, nf, ns       & 0.18\textpm0.014 & 0.19\textpm0.014 \\
        10                           & sp, nd, nf, ns       & 0.18\textpm0.011 & 0.20\textpm0.024 \\
        20                           & sp, nd, nf, ns       & 0.18\textpm0.003 & 0.19\textpm0.008 \\
        200                          & sp, nd, nf, ns       & 0.18\textpm0.011 & 0.19\textpm0.024 \\
        \bottomrule
    \end{tabular}
\end{table}

\begin{table}[H]
    \centering
    \caption{Validation dataset confusion matrix, MSE: 0.175.}
    \label{tab:oivcm}
    \begin{tabular}{lrrrrr}
        \toprule
        {} & 0   & 1  & 2  & 3  & 4   \\
        \midrule
        0  & 101 & 12 & 0  & 0  & 0   \\
        1  & 16  & 77 & 1  & 0  & 0   \\
        2  & 0   & 11 & 69 & 2  & 1   \\
        3  & 0   & 0  & 7  & 72 & 25  \\
        4  & 0   & 0  & 0  & 13 & 111 \\
        \bottomrule
    \end{tabular}
\end{table}


\subsection{Validation on new data}


\begin{figure}[H]
    \centering
    \includegraphics[width=0.9\linewidth]{p_viticola//resources//images/disease_progression_v4.png}
    \caption{Enter Caption}
    \label{fig:enter-label}
\end{figure}


\begin{table}[H]
    \centering
    \caption{ANOVA for annotated data.}
    \label{tab:anovaanno}
    \begin{tabular}{lrrrr}
        \toprule
        {}                & sum sq     & df    & F         & PR(>F)       \\
        \midrule
        C(rpvloci)        & 371.231972 & 3.0   & 64.912234 & 1.994282e-26 \\
        C(exp)            & 44.894938  & 2.0   & 11.775241 & 1.865764e-05 \\
        C(exp):C(rpvloci) & 26.338605  & 6.0   & 2.302735  & 3.758780e-02 \\
        Residual          & 268.792208 & 141.0 &           &              \\
        \bottomrule
    \end{tabular}
\end{table}

\begin{table}[H]
    \centering
    \caption{ANOVA for data predicted with binary backbone data.}
    \label{tab:anovaanno}
    \begin{tabular}{lrrrr}
        \toprule
        {}                & sum sq     & df    & F         & PR(>F)       \\
        \midrule
        C(rpvloci)        & 349.965655 & 3.0   & 95.205068 & 9.899183e-34 \\
        C(exp)            & 51.965675  & 2.0   & 21.205205 & 8.882941e-09 \\
        C(exp):C(rpvloci) & 16.967786  & 6.0   & 2.307968  & 3.718447e-02 \\
        Residual          & 172.767965 & 141.0 &           &              \\
        \bottomrule
    \end{tabular}
\end{table}

\begin{table}[H]
    \centering
    \caption{ANOVA for predicted with naive backbone data.}
    \label{tab:anovaanno}
    \begin{tabular}{lrrrr}
        \toprule
        {}                & sum sq     & df    & F          & PR(>F)       \\
        \midrule
        C(rpvloci)        & 435.641686 & 3.0   & 118.331018 & 2.490808e-38 \\
        C(exp)            & 66.281695  & 2.0   & 27.005613  & 1.176914e-10 \\
        C(exp):C(rpvloci) & 13.418454  & 6.0   & 1.822391   & 9.882385e-02 \\
        Residual          & 173.032900 & 141.0 &            &              \\
        \bottomrule
    \end{tabular}
\end{table}

\begin{table}[H]
    \centering
    \caption{Comparing Tuckey pairwise test for annotated (anno) and predicted (pred) data.}
    \label{tab:tuckey}
    \begin{tabular}{llllllll}
        \toprule
        RPV loci 1 & RPV loci 2 & meandiff anno & meandiff pred & p-adj anno & p-adj pred & reject anno & reject pred \\
        \midrule
                  & rpv1      & 2.7471        & 3.4137        & 0.0        & 0.0        & True        & True        \\
                  & rpv10     & 3.936         & 4.357         & 0.0        & 0.0        & True        & True        \\
                  & rpv3.1    & 3.7727        & 4.4906        & 0.0        & 0.0        & True        & True        \\
        rpv1      & rpv10     & 1.1889        & 0.9433        & 0.004      & 0.0101     & True        & True        \\
        rpv1      & rpv3.1    & 1.0256        & 1.0769        & 0.0166     & 0.0022     & True        & True        \\
        rpv10     & rpv3.1    & -0.1633       & 0.1336        & 0.9646     & 0.97       & False       & False       \\
        \bottomrule
    \end{tabular}
\end{table}

\section{Conclusions and perspectives}

\begin{itemize}
    \item model same result with better F score
    \item Work with various strains needs validation
    \item Will model be used next year
    \item Include annotation widget
    \item get images from other labs to feed the model
    \item What does someone else need to implement this model
\end{itemize}

\section{Supplementary}

\begin{table}[H]
    \centering
    \caption{Detail of sampling method to obtain binary phenotypes dataset.}
    \label{tab:binsample}
    \begin{tabular}{llr}
        \toprule
        sample condition & value  & quantity \\
        \midrule
        OIV              & 1      & 80 + 286       \\
        OIV              & 3      & 40 + 342      \\
        OIV              & 5      & 40 + 298     \\
        OIV              & 7      & 40 + 100 + 889   \\
        OIV              & 9      & 300 + 278     \\
        OIV \& necrosis  & 9 \& 0 & 50       \\
        high brightness  & 125    & 120      \\
        high hue         & 40     & 60       \\
        low hue          & 210    & 210      \\
        necrosis size    & 1      & 150      \\
        necrosis size    & 3      & 150      \\
        necrosis size    & 5      & 100      \\
        necrosis size    & 7      & 40       \\
        sporulation size & 1      & 55       \\
        sporulation size & 3      & 55       \\
        sporulation size & 5      & 55       \\
        random           &        & 416      \\
        riparia          &        & 100      \\
        muscadinia       &        & 100      \\   
        \bottomrule
    \end{tabular}
\end{table}

\end{document}
